### DISCUSSION TITLE
Enhance Table Row Selection with Shift+Click for Bulk Actions

### DISCUSSION THREAD

**Discussion by elbaylot** [filamentphp/filament#13070] [Ideas]
Hi,

In tables where rows are selectable and bulk actions are enabled, users currently need to select each row individually. To improve the user experience, following common UI practices, I propose implementing a feature where users can select a checkbox, hold down the Shift key, and click on another checkbox to automatically select all rows in between. This enhancement will streamline interactions and make bulk operations more efficient.

Exemple in Linear : 

https://github.com/filamentphp/filament/assets/27012416/c5463116-43a2-48f5-b307-dd3674624f3b

**Comment by neverything** [comment]
Hi @elbaylot I wanted to implement this for a while now and when I was playing around with it today, I saw your discussion. I'm now working on a PR for this https://github.com/filamentphp/filament/pull/13074. Looking forward to any feedback you might have or what I could improve upon.

**Reply by elbaylot** [reply]
Hi @neverything amazing, will check the PR.
